libraries/AP_DroneCAN/AP_DroneCAN_DNA_Server.cpp

```diff
@@ -416,6 +416,16 @@ void AP_DroneCAN_DNA_Server::handleNodeStatus(const CanardRxTransfer& transfer,
             server_state = HEALTHY;
         }
     }
+
+    //to calculate the delta b/w handlenode status pkt
+    uint32_t now = AP_HAL::native_millis();
+
+    if (_ap_dronecan.option_is_set(AP_DroneCAN::Options::LOG_NODE_STATUS)) {
+        log_NodeStatus(transfer.source_node_id, msg.uptime_sec, msg.health, msg.mode , now - last_can_init_delta_ms[transfer.source_node_id]);
+    }
+
+    last_can_init_delta_ms[transfer.source_node_id] = now;
```

Comment:
note that node IDs can be up to 127, so allocated array needs to be 128 long. Check this when you put the timestamp in the array, and move this line into if() for option